### ISSUE TITLE
Adding solutions?

### ISSUE THREAD

**Issue by ThibaultJanBeyer** [ThePrimeagen/kata-machine#19]
Hi there, would you mind also adding in the solutions for when one is stuck? I.e. the `Tries` were never implemented in the course. Cheers

**Comment by maxrzaw** [comment]
I would recommend only using solutions as a last resort. I would try wikipedia for a conceptual understanding first if that is what you are missing. That being said, I implemented a solution to Trie (and everything else) on my fork [here](https://github.com/maxrzaw/kata-machine/blob/master/src/ds/Trie.ts). I think it is similar to how ThePrimeagen implemented it when he did his speedrun. I couldn't find link since twitch removed the vod 😞. I don't have all the solutions in one place, but this might be helpful if you are still looking for a solution.

**Comment by ThibaultJanBeyer** [comment]
Ah okay, so you store the full value on the last node element that holds the last character of the string and then you return that. This increases the memory a bit I guess but makes things simpler. Thank you!

Still, I think "official" solutions would be nice in this repo to doublecheck my solution.

**Comment by maxrzaw** [comment]
Yeah, it makes it simpler at the cost of increased memory. You could also build the string as you traverse down the trie.
